yt/yt/tests/integration/queries/test_mock.py

```diff
@@ -191,3 +197,362 @@ def test_rows_limit(self, query_tracker):
         q.track()
         assert q.get_result(0)["is_truncated"]
         assert not q.get_result(1)["is_truncated"]
+
+
+class TestAccessControl(YTEnvSetup):
+    DELTA_DRIVER_CONFIG = {
+        "cluster_connection_dynamic_config_policy": "from_cluster_directory",
+    }
+
+    @authors("krock21")
+    def test_no_acos(self, query_tracker):
+        create_user("u1")
+        create_user("u2")
+        q_u1 = start_query("mock", "u1", authenticated_user="u1")
+        q_u2 = start_query("mock", "u2", authenticated_user="u2")
+        q_u1.get(authenticated_user="u1")
+        q_u2.get(authenticated_user="u2")
+        with raises_yt_error(AuthorizationErrorCode):
+            q_u1.get(authenticated_user="u2")
+        with raises_yt_error(AuthorizationErrorCode):
+            q_u2.get(authenticated_user="u1")
+
+    @authors("krock21")
+    def test_aco_doesnt_exist(self, query_tracker):
+        create_user("u1")
+        create_user("u2")
+        create_user("superuser_u3")
+        add_member("superuser_u3", "superusers")
+        create_access_control_object(
+            "aco1",
+            "queries",
+            attributes={
+                "principal_acl": [
+                    make_ace("allow", "u2", "use"),
+                ]
+            })
+        q_u1 = start_query("mock", "run_forever", authenticated_user="u1", access_control_object="aco1")
+        with raises_yt_error("Access control object \"aco2\" does not exist"):
+            start_query("mock", "run_forever", authenticated_user="u2", access_control_object="aco2")
+        q_u1.get(authenticated_user="u2")
+
+        # Delete ACO and verify that owner and superusers can still access it.
+        remove("//sys/access_control_object_namespaces/queries/aco1")
+        q_u1.get(authenticated_user="u1")
+        q_u1.get(authenticated_user="superuser_u3")
+        with raises_yt_error("Error while fetching access control object \"aco1\""):
+            q_u1.get(authenticated_user="u2")
+
+    @authors("krock21")
+    def test_aco_denies_author(self, query_tracker):
+        create_user("u1")
+        create_access_control_object(
+            "aco_denies_author",
+            "queries",
+            attributes={
+                "principal_acl": [
+                    make_ace("deny", "u1", "use"),
+                ]
+            })
+        q_u1 = start_query("mock", "run_forever", authenticated_user="u1", access_control_object="aco_denies_author")
+        q_u1.get(authenticated_user="u1")
```

Comment:
Can you please provide more motivation besides this logic? In Cypress, user can set deny themselves access to node.

Reply:
Discussed and decided that query author always has access to all actions on the query